app/queries/has_archived_coach_notes_resolver.rb

```diff
@@ -8,6 +8,10 @@ def has_archived_coach_notes # rubocop:disable Naming/PredicateName
   private
 
   def authorized?
+    return false if student&.school != current_school
```

Comment:
Why not use `resource_school`

Reply:
Please refer https://github.com/pupilfirst/pupilfirst/pull/1379#discussion_r1352514489